middleware/error.js

```diff
@@ -0,0 +1,29 @@
+const ErrorResponse = require('../utils/errorResponse');
+
+const errorHandler = (err, req, res, next) => {
+  let error = { ...err };
+
+  error.message = err.message;
+
+  // Log to console for dev
+  console.log(err.name.red);
+
+  // Mongoose bad ObjectId
+  if (err.name === 'CastError') {
+    const message = `Campaign not found with id: ${err.value}`;
```

Comment:
Since this is a middleware that will be used for all resources, shouldn't we make the error message more generic?

Reply:
will be modified in resource chapter.